vnext/proposals/active/scroll-anchoring.md

```diff
@@ -0,0 +1,100 @@
+
+#### Bottom edge anchoring
+
+When using inverted VirtualizedList, as new items are added to the "top" of the list (i.e., inserted at position 0), when at the bottom edge, the view port appears to anchor such that the newly inserted item appears to scroll into view and layout synchronously. In reality, since the ScrollView itself is flipped, the view port is just staying at offset 0.
+
+In XAML, there are two properties, [`HorizontalAnchorRatio`](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.scrollviewer.horizontalanchorratio?view=winrt-19041) and [`VerticalAnchorRatio`](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.scrollviewer.verticalanchorratio?view=winrt-19041), that implement bottom edge anchoring when given a value of 1.0 (or top edge anchoring when given a value of 0.0). Any other value controls how `IScrollAnchorProvider` chooses to anchor a specific view in the view port.
+
+One caveat to the anchor ratio properties in XAML is that you need to call `ScrollViewer.InvalidateArrange` whenever the layout of its descendants change (e.g., when a new view is inserted) in order for the edge anchoring behavior to function properly.
+
+Since currently we only want to have start or end edge anchoring, or none at all, we could add a prop called `anchorEdge` with value options `start`, `end`, or `none`.
+
+#### View anchoring not anchored at edge
+
+Another feature of inverted VirtualizedList is that view port will not jump when more content is loaded "above" it. Again, this is a side-effect of the ScrollView itself being inverted, so the view port offset actually remains the same, even though the perceived offset will have shifted downward.
+
+In XAML, we can achieve this using the [`CanBeScrollAnchor`](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.uielement.canbescrollanchor?view=winrt-19041) property on the list items. Under the hood, the XAML ScrollViewer implements `IScrollAnchorProvider`, and if a layout operation will result in a selected scroll anchor changing position, the view port will be synchronously scrolled to keep the scroll anchor in place. The `CanBeScrollAnchor` property simply tells the ScrollViewer which of its descendants are eligible to be scroll anchors.
+
+This view anchoring behavior is actually the default behavior on many Web browsers, and [some Web browsers](https://developer.mozilla.org/en-US/docs/Web/CSS/overflow-anchor#browser_compatibility) even expose a CSS property to opt-out of scroll anchoring. Since what we are trying to achieve here has effectively the same semantics as scroll anchoring on Web browsers, we could call the prop to enable scroll anchoring `overflowAnchor` and add it to `View`. Since this is not a platform default behavior, and there is likely some performance penalty to opting in to scroll anchoring, we should add an `enableAnchoring` prop to `ScrollView` to turn on scroll anchoring.
```

Comment:
I think this would be super helpful. Curious if we think this should be opt-in or opt-out for VirtualizedList.

A tangential issue I'm kind of curious about is around stability during scroll itself, instead of mutating items during mutation. I think we've seen some jank before during scrollbar thumb scroll when layout estimation is incorrect.

Reply:
@NickGerleman I think it should be opt-in. It's not platform default on iOS, macOS, Android, or Windows and likely incurs some perf penalty. For simple non-inverted FlatLists with fixed height rows, you probably don't want these to pay for anchoring by default.

We could consider adding an alternative to the "anchorEnabled" prop on ScrollView to be something like "anchorBehavior", with enum values like "anchorBehavior: explicit | implicit | none" (probably with better enum value names, explicit would mean views have to opt-in 😂. This might help address a use case @acoates-ms called out where you want to be more selective about what view is the anchor.

Reply:
@NickGerleman - FYI, I liked the idea of being able to configure whether overflow anchoring was opt-in, opt-out or neither and added it to the proposal :)